extensions/guacamole-auth-restrict/src/main/java/org/apache/guacamole/calendar/TimeRestrictionParser.java

```diff
@@ -0,0 +1,135 @@
+     */
+    private static final int RESTRICTION_TIME_END_GROUP = 4;
+    
+    /**
+     * Parse the provided string containing one or more restrictions into
+     * a list of objects.
+     * 
+     * @param restrictionString
+     *     The string that should contain one or more semicolon-separated
+     *     restriction periods.
+     * 
+     * @return 
+     *     A list of objects parsed from the string.
+     */
+    public static List<DailyRestriction> parseString(String restrictionString) {
+        
+        List<DailyRestriction> restrictions = new ArrayList<>();
+        Matcher restrictionMatcher = RESTRICTION_REGEX.matcher(restrictionString);
+        
+        // Loop through RegEx matches
+        while (restrictionMatcher.find()) {
+            
+            // Pull the day string, start time, and end time
+            String dayString = restrictionMatcher.group(RESTRICTION_DAY_GROUP);
+            String startTimeString = restrictionMatcher.group(RESTRICTION_TIME_START_GROUP);
+            String endTimeString = restrictionMatcher.group(RESTRICTION_TIME_END_GROUP);
+            int dayStart, dayEnd;
+            
+            // We must always have a start day.
+            if (dayString == null || dayString.isEmpty())
+                continue;
+            
+            // Check to see if the day string contains a range or not.
+            if (dayString.contains("-")) {
+                String dayStrings[] = dayString.split("-");
+                dayStart = Integer.parseInt(dayStrings[0]);
+                dayEnd = Integer.parseInt(dayStrings[1]);
+            }
+            else {
+                dayStart = Integer.parseInt(dayString);
+                dayEnd = Integer.parseInt(dayString);
+            }
+            
+            // Convert the start and end time strings to LocalTime values.
+            DateTimeFormatter hourFormat = DateTimeFormatter.ofPattern("HHmm");
+            LocalTime startTime = LocalTime.parse(startTimeString, hourFormat);
+            LocalTime endTime = LocalTime.parse(endTimeString, hourFormat);
+            
+            // Loop through days and add entries for each day
+            for (int i = dayStart; i <= dayEnd; i++) {
+                DayOfWeek weekDay = DayOfWeek.of(i);
+                
+                // We must always have both start time and end time.
+                if (startTimeString == null || startTimeString.isEmpty() 
+                        || endTimeString == null || endTimeString.isEmpty())
+                    restrictions.add(new DailyRestriction(weekDay));
+
+                else
+                    restrictions.add(new DailyRestriction(weekDay, 
+                            startTime, endTime));
+            }
+            
+        }
+        
+        // Return the list of restrictions
+        return restrictions;
+        
+    }
+    
+    public static String toString(List<DailyRestriction> restrictions) {
```

Comment:
Is this used? If so it needs to be documented.

Reply:
Nope, not used - removed it.